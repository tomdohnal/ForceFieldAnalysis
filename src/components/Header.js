// @flow
import React, { Component } from 'react';
import { Divider } from 'semantic-ui-react';

import { Box, Text } from './common';
import { COLORS, HEADER_HEIGHT } from '../constants/styles';

type State = {
  showAbout: boolean,
}

class Header extends Component<any, State> {
  state = {
    showAbout: false,
  };

  onAboutLinkClick = () => {
    this.setState({ showAbout: true });
  };

  onCloseAboutLinkClick = () => {
    this.setState({ showAbout: false });
  };

  render() {
    return (
      <Box height={HEADER_HEIGHT} paddingTop="16px" textAlign="center" position="relative">
        <Text fontSize="24px">Force Field Analysis Tool</Text>
        {this.state.showAbout ?
          <Box
            position="absolute"
            zIndex="999"
            backgroundColor={COLORS.WHITE}
            width="500px"
            right="16px"
            padding="16px"
            textAlign="initial"
            border={`2px ${COLORS.BLUE} solid`
            }
          >
            <Text
              onClick={this.onCloseAboutLinkClick}
              position="absolute"
              right="16px"
              top="8px"
              clickable
            >Close
            </Text>
            <Text fontSize="32px" marginTop="16px">What is Force Field Analysis?</Text>
            <Divider hidden />
            <Text fontSize="18px">Force Field Analysis is a decision-making tool. Originally created in 1940s by psychologist Kurt Lewin, it is widely used in business for making decisions. The idea is to create driving (also called helping) forces that are driving change and hindering (also called resisting) forces that are hindering change. You can attribute different strengths to each force. Once you've done with that, you can compare the sum of your driving forces and your hindering forces and decide if you should go or not go for change!</Text>
          </Box>
          :
          <Text
            onClick={this.onAboutLinkClick}
            position="absolute"
            right="20px"
            clickable
          >About
          </Text>
        }
      </Box>
    );
  }
}

export default Header;
