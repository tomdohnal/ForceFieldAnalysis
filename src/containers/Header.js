// @flow
import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Button, Grid } from 'semantic-ui-react';

import { Box, Text } from '../components/common/index';
import { COLORS, HEADER_HEIGHT } from '../constants/styles';
import AboutBox from '../components/AboutBox';
import { resetState, type ResetStateAction, type ReduxState } from '../redux';
import { switchToCompleteMode, switchToEditMode, type SwitchToCompleteModeAction, type SwitchToEditModeAction, type Mode, EDIT_MODE } from '../ducks/mode';

type Props = {
  resetState: () => ResetStateAction,
  switchToCompleteMode: () => SwitchToCompleteModeAction,
  switchToEditMode: () => SwitchToEditModeAction,
  appMode: Mode,
}

type State = {
  showAbout: boolean,
}

export class Header extends Component<Props, State> {
  state = {
    showAbout: false,
  };

  onResetButtonClick = (e: SyntheticEvent<HTMLButtonElement>) => {
    this.props.resetState();

    if (e.target instanceof HTMLButtonElement) {
      e.target.blur();
    }
  };

  onMakeDecisionButtonClick = (e: SyntheticEvent<HTMLButtonElement>) => {
    this.props.switchToCompleteMode();

    if (e.target instanceof HTMLButtonElement) {
      e.target.blur();
    }
  };

  onEditForcesButtonClick = (e: SyntheticEvent<HTMLButtonElement>) => {
    this.props.switchToEditMode();

    if (e.target instanceof HTMLButtonElement) {
      e.target.blur();
    }
  };

  onAboutLinkClick = () => {
    this.setState({ showAbout: true });
  };

  onCloseAboutLinkClick = () => {
    this.setState({ showAbout: false });
  };

  render() {
    return (
      <Box height={HEADER_HEIGHT} paddingTop="16px">
        <Grid columns={3}>
          <Grid.Row>
            <Grid.Column>
              <Box paddingLeft="16px">
                <Button
                  onClick={this.onResetButtonClick}
                  style={{ color: COLORS.WHITE, background: COLORS.RED }}
                >
                  Reset analysis
                </Button>
                {this.props.appMode === EDIT_MODE ?
                  <Button onClick={this.onMakeDecisionButtonClick}>Make decision</Button>
                  :
                  <Button onClick={this.onEditForcesButtonClick}>Edit forces</Button>
                }
              </Box>
            </Grid.Column>
            <Grid.Column>
              <Box textAlign="center">
                <Text fontSize="24px">Force Field Analysis Tool</Text>
              </Box>
            </Grid.Column>
            <Grid.Column floated="right">
              {this.state.showAbout ?
                <AboutBox onCloseLinkClick={this.onCloseAboutLinkClick} />
                :
                <Text
                  onClick={this.onAboutLinkClick}
                  position="absolute"
                  right="30px"
                  clickable
                >About
                </Text>
              }
            </Grid.Column>
          </Grid.Row>
        </Grid>
      </Box>
    );
  }
}

export const mapStateToProps = (state: ReduxState) => ({ appMode: state.mode });

export default connect(
  mapStateToProps, { resetState, switchToCompleteMode, switchToEditMode },
)(Header);
