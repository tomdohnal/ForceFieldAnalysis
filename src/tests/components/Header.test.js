import React from 'react';
import { shallow } from 'enzyme';

import Header from '../../components/Header';

describe('Header', () => {
  it('renders correctly', () => {
    const tree = shallow(<Header />);

    expect(tree).toMatchSnapshot();
  });
});
