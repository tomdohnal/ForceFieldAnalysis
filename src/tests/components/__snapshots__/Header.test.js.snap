// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<boxStyles
  height="60px"
  paddingTop="16px"
  position="relative"
  textAlign="center"
>
  <text
    fontSize="24px"
  >
    Force Field Analysis Tool
  </text>
  <text
    clickable={true}
    onClick={[Function]}
    position="absolute"
    right="20px"
  >
    About
  </text>
</boxStyles>
`;
